lib/create-host-config.ts

```diff
@@ -268,21 +285,59 @@ export function createHostConfig(jscad: JSCADModule) {
 
       case "subtract": {
         const { children } = props as SubtractProps
-        if (!Array.isArray(children) || children.length < 2) {
-          throw new Error("Subtract must have at least two children")
+
+        if (!children || children.length < 2) {
+          throw new Error(
+            "Subtract must have at least one base component and one component to subtract.",
+          )
         }
 
-        const geometries = children.map((child: any) =>
-          createInstance(
-            child.type,
-            child.props,
-            rootContainerInstance,
-            hostContext,
-            internalInstanceHandle,
-          ),
+        // Filter to only include valid React elements
+        const validChildren = React.Children.toArray(children).filter(
+          React.isValidElement,
+        )
+
+        if (validChildren.length < 2) {
+          throw new Error(
+            "Subtract must have at least one base component and one component to subtract.",
+          )
+        }
+
+        // Convert the base component (first child) to JSCAD geometry
+        const baseGeometry = createInstance(
+          validChildren[0].type,
+          validChildren[0].props,
+          rootContainerInstance,
+          hostContext,
+          internalInstanceHandle,
+        )
+
+        // Convert the rest of the valid children (subtraction components) to JSCAD geometries
+        const subtractGeometries = flattenArray(
+          validChildren
+            .slice(1)
+            .map((child) =>
+              createInstance(
+                child.type,
+                child.props,
+                rootContainerInstance,
+                hostContext,
+                internalInstanceHandle,
+              ),
+            ),
         )
-        return geometries.reduce((acc, curr) =>
-          jscad.booleans.subtract(acc, curr),
+
+        // Ensure the base and subtract geometries were created successfully
+        if (!baseGeometry || subtractGeometries.some((geo) => geo == null)) {
+          throw new Error(
+            "One or more geometries could not be processed for subtraction.",
+          )
+        }
+
+        // Apply JSCAD's subtract operation across the array of subtraction geometries
+        return subtractGeometries.reduce(
+          (acc, curr) => jscad.booleans.subtract(acc, curr),
```

Comment:
We shouldnt need to use reduce

Reply:
I think you should be able to pass multiple parameters to subtract, the reduce operation is very expensive relatively

Reply:
The documentation states you can pass multiple geometries in (eg as a spread parameter)

![image](https://github.com/user-attachments/assets/c6280d26-b848-4083-854f-90161baa67ec)

Reply:
Cool lets refactor to use that instead of a reduce here?